ocaml/alerts/certificate/certificate_check.ml

```diff
@@ -38,6 +38,10 @@ let generate_alert epoch (host, expiry) =
     host, Some (message, alert)
 
 let execute rpc session (host, alert) =
+  if Xapi_stdext_unix.Unixext.string_of_file "/etc/xensource/pool.conf" <> "master" then (
+    raise (Failure "Alerts cannot only be executed from a master host")
```

Comment:
Well, they can, but we want to do it in the cron job of just one host. I would therefore put this check into the cron script, which can just exit if not the master.